File: typescript/src/registration_search.ts
import { createNodeClient } from "./client"
import { create, toJsonString } from "@bufbuild/protobuf";
import {
  RegistrationSearchRequestSchema,
  RegistrationSearchResponseSchema,
  RegistrationRegion
} from "@buf/nativeconnect_api.bufbuild_es/nativeconnect/api/v1/nativeconnect_pb";

const client = createNodeClient()

const request = create(RegistrationSearchRequestSchema, {
  registration: {
    plateNumber: "ABC123",
    registrationRegion: RegistrationRegion.REGISTRATION_AUS_NSW
  }
})

const response = await client.registrationSearch(request)

console.log(toJsonString(RegistrationSearchResponseSchema, response, { prettySpaces: 2 }))

File: python/registration_search.py
from client import create_native_client
import nativeconnect.api.v1.nativeconnect_pb2 as native


def main():
    client = create_native_client()

    request = native.RegistrationSearchRequest(
        registration=native.Registration(
            plate_number="ABC123",
            registration_region=native.RegistrationRegion.REGISTRATION_AUS_NSW
        )
    )

    response = client.RegistrationSearch(request)

    print(response)


if __name__ == "__main__":
    main()
